src/rational/mat_q/cmp.rs

```diff
@@ -116,3 +178,33 @@ mod test_partial_eq {
         assert_ne!(&d, &e);
     }
 }
+
+/// Test that the [`PartialEq`] trait is correctly implemented.
+#[cfg(test)]
+mod test_partial_eq_q_other {
+    use super::MatQ;
+    use crate::integer::MatZ;
+    use std::str::FromStr;
+
+    // Ensure that the function can be called with several types
+    #[test]
+    #[allow(clippy::op_ref)]
+    fn availability() {
+        let q = MatQ::from_str("[[1, 2],[3, 4]]").unwrap();
+        let z = MatZ::from_str("[[1, 2],[3, 4]]").unwrap();
+
+        assert!(q == z);
+        assert!(z == q);
+        assert!(&q == &z);
+        assert!(&z == &q);
+    }
+
+    // Ensure that large values are compared correctly
+    #[test]
+    fn equal_large() {
```

Comment:
```suggestion
    /// Ensure that large values are compared correctly.
    #[test]
    fn equal_large() {
```
What about inequalitites? A quick test for that would be nice.